src/Data/Masa.BuildingBlocks.Data.Mapping/MapperConfiguration.cs

```diff
@@ -0,0 +1,23 @@
+﻿// Copyright (c) MASA Stack All rights reserved.
+// Licensed under the MIT License. See LICENSE.txt in the project root for license information.
+
+namespace Masa.BuildingBlocks.Data.Mapping;
+
+public class MapperConfiguration
+{
+    private readonly IServiceCollection _services;
+
+    internal IMapperOptionsExtension Extension { get; set; } = default!;
+
+    public static MapperConfiguration Instance = new();
+
+    public MapperConfiguration() => _services = new ServiceCollection();
+
+    public void ConfigurationMapperOptions(IMapperOptionsExtension extension) => Extension = extension;
+
+    public Mapper CreateMapper()
+    {
+        Extension.AddService(_services);
```

Comment:
Why call AddService everytime?

Reply:
There is no case of adding multiple times, it is only executed once during initialization, the following is the demo

initialization:

``` C#
MapperFactory.Instance.AddMapping().CreateMapper();
```

use:
``` C#
var request = new CreateUserRequest()
{
    Name = "Jim",
};
var user = request.Map<CreateUserRequest, User>();
Assert.IsNotNull(user);
Assert.IsTrue(user.Name == request.Name);
```

Reply:
It will indeed be executed once more in processing